src/Microsoft.Extensions.Localization/Internal/IResourceStringManager.cs

```diff
@@ -0,0 +1,13 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved. 
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information. 
+
+using System.Collections.Generic;
+using System.Globalization;
+
+namespace Microsoft.Extensions.Localization.Internal
+{
+    public interface IResourceStringProvider
+    {
+        IList<string> GetAllResourceStrings(CultureInfo culture, bool throwOnMissing);
```

Comment:
That would require us to change `IResourceNamesCache`, which is a public interface.